File: ABC/241-250/248/D.cc
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef long double ld;
#define _rep(i, a, b) for (ll i = a, i##_len = (ll)(b); i < i##_len; i++)
#define rep(i, n) _rep(i,0,n)
#define reps(i, n) _rep(i,1,n+1)
#define rrep(i, n) for (ll i = n-1; i >= 0; i--)
#define rreps(i, n) for (ll i = n; i >= 1; i--)
#define YesNo(flag) if(flag){cout<<"Yes"<<endl;}else{cout<<"No"<<endl;}
#define _10_5 100010
#define _2_10_5 200010
#define mod 998244353
#define mod2 1000000007
#define inf 1000000000
#define INF 1000000000000000000L
// sort(ALL(lst),[](const vector<int> &alpha,const vector<int> &beta){return alpha[N] < beta[N];}); <- N番目の要素でソート



int main(void){
    ll n;
    cin >> n;
    vector<vector<long long>> index(n+1);
    ll m;
    rep(i,n){
        cin >> m;
        index[m].push_back(i);
    }
    ll Q;
    cin >> Q;
    ll L,R,X;
    rep(i,Q){
        cin >> L >> R >> X;
        cout << upper_bound(index[X].begin(), index[X].end(), R) - lower_bound(index[X].begin(), index[X].end(), L) << endl;
    }
}

File: ABC/241-250/248/D.py
import sys
inputSys = sys.stdin.readline
# sys.setrecursionlimit(10**7)
import copy  # copy.deepcopy()
import itertools  # combinations() <- nCr, permutations() <- nPr
import bisect  # 二分探索
import math  # factorial() <- 階乗, sqrt() <- 平方根
from functools import lru_cache # <- メモ化再帰 @lru_cache
import string
mod = 998244353
mod2 = 1000000007
inf = float("inf")
minf = -float("inf")
def YesNo(flag: bool, yes="Yes", no="No"): print(yes) if flag else print(no)
# lst.sort(key=lambda x: x[N]) <- N番目の要素でソート
# sorted_lst = sorted(lst.items(), key=lambda x: x[0]) <- 辞書のkeyでソート
# sorted_lst = sorted(lst.items(), key=lambda x: x[1]) <- 辞書のvalueでソート



n = int(input())
lst = list(map(int, inputSys().rstrip().split()))
indexs = [[] for _ in range(n+1)]
for i, val in enumerate(lst):
    indexs[val].append(i)
Q = int(inputSys())

for i in range(Q):
    L, R, X = map(int, inputSys().rstrip().split())
    print(bisect.bisect_right(indexs[X], R-1) - bisect.bisect_left(indexs[X], L-1))
